content/tutorial-react-native/rn-03.md

```diff
@@ -0,0 +1,343 @@
+do the extra request to show yout how to customize the request based on the route.
+
+We already created the navigation in `navigation/` that assigns the `PokemonDetailScreen` to the path `pokemonDetail` 
+so we can. Exponent navigation allows us to pass extra parameters to the route like this :
+
+```js
+_goToDetail = (pokemon) => {
+  this.props.navigation
+    .getNavigator('root')
+    .push(Router.getRoute('pokemonDetail', { pokemon }));
+}
+```
+
+where `getNavigator` allows to select the root navigator (and not the tab navigator) and push some routes on it, which makes
+the second order pages'tab bar disappear, and `push` allow to pass an extra object of parameters.
+
+Then we can get the paramaters like this :
+
+```js
+<PokemonDetailWithData pokemonId={this.props.route.params.pokemon.id}/>
+}
+```
+
+
+The `PokemonDetailScreen` component is responsible to pass down a pokemon to the `PokemonDetail` component. Let's add a 
+`PokemonQuery` to `PokemonDetail` that is fetching the required pokemon object:
+
+```js
+const PokemonQuery = gql`
+  query PokemonQuery($id: ID!) {
+    Pokemon(id: $id) {
+      id
+      url
+      name
+    }
+  }
+`
+```
+
+As you can see, the query requires a query variable `id` of type `ID` that we have to supply using the query option 
+`variables` as before. However, other than with the trainer name variable, we cannot just use a cannot value as the 
+id in our case. For that reason, we have the possibility to access the props when creating the `variables` object. 
+So, we can replace:
+
+```js
+export PokemonDetail;
+```
+
+with:
+
+```js
+export const PokemonDetailWithData = graphql(PokemonQuery, {
+  options: (ownProps) => {
+    return {
+      variables: {
+        id: ownProps.pokemonId
+      }
+    }
+  }
+})(PokemonDetail);
+```
+
+Note that by default Apollo try to fill unfilled `variable` with `ownProps.variable`, so this was valid too:
+
+
+```js
+const PokemonQuery = gql`
+  query PokemonQuery($pokemonId: ID!) {
+    Pokemon(id: $pokemonId) {
+      id
+      url
+      name
+    }
+  }
+`
+export const PokemonDetailWithData = graphql(PokemonQuery)(PokemonDetail);
+```
+
+To confirm your environment is all correctly setup, start the app now from XDE. 
+
+You should see a list of pokemons. Click on a pokemon preview to move over to the detailed view. 
+Click the back button on the nav bar or the native back button on Android to go back to your pokedex.
+
+## Recap
+
+Nice, our pokedex starts to get some shape! We added both an overview of all our pokemons as well as a detailed view of 
+a single pokemon. Let's go through what we saw in this exercise again:
+
+* **Constants as query variables** helped us getting started with advanced query features
+* Using GraphQL, we can easily create **nested queries**
+* Combining **router parameters** and **dynamic query variables**, we were able to supply different data to the same 
```

Comment:
is this still relevant? maybe customize this line to the exponent navigation stuff